src/clj/puppetlabs/http/client/sync.clj

```diff
@@ -23,43 +23,63 @@
   (select-keys opts [:url :method :headers :body :decompress-body :as :query-params]))
 
 (defn request-with-client
-  [req client]
-  (let [{:keys [error] :as resp} @(async/request-with-client req nil client)]
+  [req client metric-registry]
+  (let [{:keys [error] :as resp} @(async/request-with-client req nil client metric-registry)]
     (if error
       (throw error)
       resp)))
 
+(defn get-client-metrics
+  ([metric-registry]
+   (async/get-client-metrics metric-registry))
+  ([metric-registry metric-filter]
+   (async/get-client-metrics metric-registry metric-filter)))
+
+(defn get-client-metrics-data
+  ([metric-registry]
+   (async/get-client-metrics-data metric-registry))
+  ([metric-registry metric-filter]
+   (async/get-client-metrics-data metric-registry metric-filter)))
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
 (defn request
   [req]
   (with-open [client (async/create-default-client (extract-client-opts req))]
-    (request-with-client (extract-request-opts req) client)))
+    (request-with-client (extract-request-opts req) client nil)))
```

Comment:
This is not specifically in response to the question of whether we should track metrics for one-off requests, but that is what got me thinking about it. The more I think about it, the more I wonder whether it would make a lot more sense to have the `MetricRegistry` passed in as a client option, rather than as its own argument to `create-client`. No other client option gets is own argument, and I think the only reason that I wrote the code this way to begin with is that I didn't quite understand how the client options worked.